framework/src/main/java/org/checkerframework/framework/type/SupertypeFinder.java

```diff
@@ -216,28 +216,50 @@
      */
     private Map<TypeVariable, AnnotatedTypeMirror> getTypeVarToTypeArg(AnnotatedDeclaredType type) {
       Map<TypeVariable, AnnotatedTypeMirror> mapping = new HashMap<>();
-      AnnotatedDeclaredType enclosing = type;
-      while (enclosing != null) {
-        TypeElement enclosingTypeElement = (TypeElement) enclosing.getUnderlyingType().asElement();
-        List<? extends TypeParameterElement> typeParams = enclosingTypeElement.getTypeParameters();
-        List<AnnotatedTypeMirror> typeArgs = enclosing.getTypeArguments();
-        for (int i = 0; i < enclosing.getTypeArguments().size(); ++i) {
-          AnnotatedTypeMirror typArg = typeArgs.get(i);
-          TypeParameterElement ele = typeParams.get(i);
-          mapping.put((TypeVariable) ele.asType(), typArg);
-        }
+      // addTypeVarsFromEnclosingTypes can't be called with `type` because it calls
+      // `directSupertypes(types)`, which then calls this method. Add the type variables from `type`
+      // and then recur on the enclosing type and super types.
+      addTypeVariablesToMapping(mapping, type);
+      addTypeVarsFromEnclosingTypes(type.getEnclosingType(), mapping);
+      return mapping;
+    }
 
-        @SuppressWarnings("interning:not.interned") // First time through type == enclosing.
-        boolean notType = enclosing != type;
-        if (notType) {
-          for (AnnotatedDeclaredType enclSuper : directSupertypes(enclosing)) {
-            mapping.putAll(getTypeVarToTypeArg(enclSuper));
-          }
-        }
+    /**
+     * Adds a mapping from a type parameter to its corresponding annotated type argument for all
+     * type parameters of {@code type}.
+     *
+     * @param mapping type variable to type argument map; side-effected by this method
+     * @param type a type
+     */
+    private void addTypeVariablesToMapping(
+        Map<TypeVariable, AnnotatedTypeMirror> mapping, AnnotatedDeclaredType type) {
+      TypeElement enclosingTypeElement = (TypeElement) type.getUnderlyingType().asElement();
+      List<? extends TypeParameterElement> typeParams = enclosingTypeElement.getTypeParameters();
+      List<AnnotatedTypeMirror> typeArgs = type.getTypeArguments();
+      for (int i = 0; i < type.getTypeArguments().size(); ++i) {
+        AnnotatedTypeMirror typArg = typeArgs.get(i);
+        TypeParameterElement ele = typeParams.get(i);
+        mapping.put((TypeVariable) ele.asType(), typArg);
+      }
+    }
 
+    /**
+     * Creates a mapping from a type parameter to its corresponding annotated type argument for all
```

Comment:
This modifies an existing mapping rather than creating a mapping.  (Documentation seems to have been copied from `getTypeVarToTypeArg()`.)